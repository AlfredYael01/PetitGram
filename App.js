import React, { useEffect } from 'react';
import { View, Text } from 'react-native';
import MainScreen from './components/Main';
import { Provider, useSelector, useDispatch } from 'react-redux';
import store from './components/redux/store';
import { getApps, initializeApp } from 'firebase/app';
import { onAuthStateChanged, getAuth, signInWithEmailAndPassword, createUserWithEmailAndPassword } from 'firebase/auth';
import { setUser } from './components/redux/userActions';

const firebaseConfig = {
    apiKey: "AIzaSyD4IOL2vqYSiGP4l8Icg_uCAmNo4mq4qU0",
    authDomain: "petitgram-b48fd.firebaseapp.com",
    projectId: "petitgram-b48fd",
    storageBucket: "petitgram-b48fd.appspot.com",
    messagingSenderId: "425299538465",
    appId: "1:425299538465:web:3356a2599e6e679b4d6960",
    measurementId: "G-RS2NF646G0"
};
// Initialize Firebase
if (getApps().length === 0) {
  initializeApp(firebaseConfig);
}

const email = 'test@gmail.com';
const password = 'password';

function App() {
  const dispatch = useDispatch();
  const user = useSelector((state) => state.user);

  useEffect(() => {
    // Create user with email and password if not exists
    const auth = getAuth();

    signInWithEmailAndPassword(auth, email, password)
      .then((userCredential) => {
        // Signed in
        const user = userCredential.user;
        dispatch(setUser(user)); // Dispatch user data to Redux
        // ...
      })
      .catch((error) => {
        const errorCode = error.code;
        const errorMessage = error.message;
        console.log(error);
        // ..
      });

    const subscriber = onAuthStateChanged(auth, (user) => {
      if (user) {
        dispatch(setUser(user)); // Dispatch user data to Redux
      }
    });

    return subscriber; // Unsubscribe on unmount
  }, [dispatch]);

  if (!user) {
    return (
      <View>
        <Text>Login</Text>
      </View>
    );
  }

  return (
    <MainScreen />
  );
}

export default function ReduxApp() {
  return (
    <Provider store={store}>
      <App />
    </Provider>
  );
}
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',  
  }

});
