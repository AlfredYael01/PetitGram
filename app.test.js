import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import TestRenderer from 'react-test-renderer';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);

// mock firebase/auth module
jest.mock('firebase/auth', () => ({
  ...jest.requireActual('@firebase/auth'),
  getReactNativePersistence: () => console.debug('Initialized persistence ...'),
}));

describe('<App />', () => {
  it('login renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});