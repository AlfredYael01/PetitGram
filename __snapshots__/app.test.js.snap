// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> login renders correctly 1`] = `
[
  <View />,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#F8F8F8",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 250,
          "marginBottom": 0,
          "width": 250,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "elevation": 2,
          "justifyContent": "center",
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, 0.2)",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 2,
          "width": "80%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Email"
        style={
          {
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 40,
            "marginVertical": 10,
            "padding": 10,
            "width": "100%",
          }
        }
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          {
            "borderRadius": 5,
            "borderWidth": 1,
            "height": 40,
            "marginVertical": 10,
            "padding": 10,
            "width": "100%",
          }
        }
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign In
          </Text>
        </View>
      </View>
    </View>
    <Text
      onPress={[Function]}
      style={
        {
          "color": "#007BFF",
          "marginTop": 20,
          "textDecorationLine": "underline",
        }
      }
    >
      Don't have an account? Sign Up.
    </Text>
  </View>,
]
`;
